File: utility/src/main/java/org/oppia/util/parser/ImageLoader.java
package org.oppia.util.parser;

import android.content.Context;
import android.graphics.Bitmap;
import androidx.annotation.DrawableRes;
import com.bumptech.glide.Glide;
import com.bumptech.glide.request.target.SimpleTarget;

import java.lang.annotation.Target;

public interface ImageLoader {

  static void load(Context context, String path, int placeholder, SimpleTarget<Bitmap> target) {
    Glide.with(context)
        .asBitmap()
        .load(path)
        .placeholder(placeholder)
        .into(target);
  }
}

File: utility/src/main/java/org/oppia/util/parser/ImageLoader.kt
package org.oppia.util.parser

import android.graphics.Bitmap
import android.graphics.Picture
import com.bumptech.glide.request.target.CustomTarget

/** Loads an image from the provided URL into the specified target, optionally caching it. */
interface ImageLoader {

  fun load(imageUrl: String, target: CustomTarget<Bitmap>)

  fun loadSvg(imageUrl: String, target: CustomTarget<Picture>)
}
